### PR TITLE
BACK-1141 : LWD/PubKey Authentication removal

### PR DESCRIPTION
**Pull request by RomanWlm**
Remove Authentication and User entity
WalletPool Instances are cached, several WalletPool instances trigger libcore crashes.

### CODE REVIEW

File: src/it/scala/co/ledger/wallet/daemon/utils/APIFeatureTest.scala [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
 import com.twitter.finatra.http.EmbeddedHttpServer
 import com.twitter.inject.server.FeatureTest
-import org.bitcoinj.core.Sha256Hash
 
 trait APIFeatureTest extends FeatureTest {
   val serverImpl = new ServerImpl
   override val server = new EmbeddedHttpServer(serverImpl)
 
-  def defaultHeaders: Map[String, String] = lwdBasicAuthorisationHeader("whitelisted")
+  def defaultHeaders: Map[String, String] = Map.empty
```
Comment (mbasire-ledger) [inline src/it/scala/co/ledger/wallet/daemon/utils/APIFeatureTest.scala:13]: It should be a val. BTW, is `headers` parameter mandatory ?

**Review by mbasire-ledger — CHANGES_REQUESTED** [review]
CI Integration tests do not pass.

File: src/main/scala/co/ledger/wallet/daemon/database/DBMigrations.scala [hunk 2]
```diff
@@ -8,6 +8,6 @@ object DBMigrations {
   import Tables._
   import Tables.profile.api._
   val Migrations = Map(
-    0 -> (users.schema ++ pools.schema ++ databaseVersions.schema).create
+    0 -> (pools.schema ++ databaseVersions.schema).create
```
[the remark this answers is not in the document]
Comment (RomanWlm) [reply src/main/scala/co/ledger/wallet/daemon/database/DBMigrations.scala:11]: No I tested on a existing db it works, the reverse doesnot

File: src/it/scala/co/ledger/wallet/daemon/utils/APIFeatureTest.scala (hunk 1 above)
Comment (RomanWlm) [reply src/it/scala/co/ledger/wallet/daemon/utils/APIFeatureTest.scala]: No, just a safe shortcut that could be removed IMO

**Pull request merged**